central/sensor/service/connection/connection_impl.go

```diff
@@ -603,19 +605,37 @@ func (c *sensorConnection) Run(ctx context.Context, server central.SensorService
 		log.Infof("Sensor (%s) can do client reconciliation: sending deduper state", c.clusterID)
 
 		// Send hashes to sensor
+		maxEntries := env.MaxDeduperEntriesPerMessage.IntegerSetting()
 		successfulHashes := c.hashDeduper.GetSuccessfulHashes()
-		deduperMessage := &central.MsgToSensor{Msg: &central.MsgToSensor_DeduperState{
-			DeduperState: &central.DeduperState{
-				ResourceHashes: successfulHashes,
-			},
-		}}
-
-		log.Infof("Sending %d hashes (Size=%d)", len(successfulHashes), deduperMessage.Size())
-
-		err := server.Send(deduperMessage)
-		if err != nil {
-			log.Errorf("Central wasn't able to send deduper state to sensor (%s): %s", c.clusterID, err)
-			return errors.Wrap(err, "unable to sync deduper state")
+		// If there are no hashes we send the empty map
+		if len(successfulHashes) == 0 {
+			if err := c.sendDeduperState(server, successfulHashes, 1, 1); err != nil {
+				return err
+			}
+		}
+		// Since the environment variable can be set to zero or some negative number.
+		// Make sure maxEntries is a valid number to not panic
+		if maxEntries < 1 {
+			maxEntries = 200000
+		}
+		log.Debugf("DeduperState chunk size %d", maxEntries)
```

Comment:
nit: move this log one line below and include values of `len(successfulHashes)` and `total`

Reply:
Done